src/Features/Core/Portable/SignatureHelp/SignatureHelpItems.cs

```diff
@@ -57,24 +61,17 @@ internal class SignatureHelpItems
     public SignatureHelpItems(
         IList<SignatureHelpItem> items,
         TextSpan applicableSpan,
-        int argumentIndex,
-        int argumentCount,
+        int semanticParameterIndex,
+        int syntacticArgumentCount,
         string? argumentName,
         int? selectedItem = null)
     {
         Contract.ThrowIfNull(items);
         Contract.ThrowIfTrue(items.IsEmpty());
         Contract.ThrowIfTrue(selectedItem.HasValue && selectedItem.Value >= items.Count);
 
-        if (argumentIndex < 0)
-        {
-            throw new ArgumentException($"{nameof(argumentIndex)} < 0. {argumentIndex} < 0", nameof(argumentIndex));
-        }
-
-        if (argumentCount < argumentIndex)
-        {
-            throw new ArgumentException($"{nameof(argumentCount)} < {nameof(argumentIndex)}. {argumentCount} < {argumentIndex}", nameof(argumentIndex));
-        }
```

Comment:
this was the crux of hte bug.  a check that literally made no sense.  the number of syntactic arguments provided has no basis on which semantic *parameter* we think is teh right one to select.  It's just information about the syntax tree, that's all.